### PR TITLE
[iOS 16] Simplify test infrastructure around observing edit menu appearance and dismissal

### PR DESCRIPTION
**Pull request by whsieh**
#### 8641efca8a2287a1286e82c5ecbff242e42fd3eb
<pre>
[iOS 16] Simplify test infrastructure around observing edit menu appearance and dismissal
<a href="https://bugs.webkit.org/show_bug.cgi?id=243101">https://bugs.webkit.org/show_bug.cgi?id=243101</a>

Reviewed by Tim Horton and Megan Gardner.

We currently swizzle some UIKit delegate method implementations (on internal UIKit objects) in order
to observe when the edit menu is presented or dismissed, for layout tests. UIKit just recently added
new API in rdar://94043113 which makes it possible to achieve this without having to swizzle
anything; we should adopt this API to:

1. Simplify our test harness logic around observing the edit menu, and
2. Make this test harness logic more robust, against future changes in UIKit.

* Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:

In order achieve this, I&apos;m also adding this delegate plumbing as SPI instead of just internal
`WKWebView` testing hooks, since we intend to make this public API soon in rdar://96239605; as such,
this patch also works towards that radar.

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView willPresentEditMenuWithAnimator:]):
(-[WKContentView willDismissEditMenuWithAnimator:]):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.h:
* Tools/WebKitTestRunner/cocoa/TestRunnerWKWebView.mm:
(-[TestRunnerWKWebView _webView:willPresentEditMenuWithAnimator:]):
(-[TestRunnerWKWebView _webView:willDismissEditMenuWithAnimator:]):

Implement the new UI delegate hooks by calling into `-_didShowMenu` and `-_didHideMenu`.

(-[TestRunnerWKWebView immediatelyDismissEditMenuInteractionIfNeeded]):
(-[TestRunnerWKWebView currentEditMenuInteraction]): Deleted.
(-[TestRunnerWKWebView didPresentEditMenuInteraction:]): Deleted.
(-[TestRunnerWKWebView didDismissEditMenuInteraction:]): Deleted.
* Tools/WebKitTestRunner/ios/EditMenuInteractionSwizzler.h: Removed.
* Tools/WebKitTestRunner/ios/EditMenuInteractionSwizzler.mm: Removed.

Remove this entire swizzler helper class, since it&apos;s no longer necessary.

* Tools/WebKitTestRunner/ios/TestControllerIOS.mm:
(WTR::TestController::platformInitialize):
(WTR::TestController::didPresentEditMenuInteraction): Deleted.
(WTR::TestController::didDismissEditMenuInteraction): Deleted.

Canonical link: <a href="https://commits.webkit.org/252742@main">https://commits.webkit.org/252742@main</a>
</pre>

### CODE REVIEW

File: Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h [hunk 1]
```diff
@@ -255,6 +256,9 @@ struct UIEdgeInsets;
 - (NSArray<UIDragItem *> *)_webView:(WKWebView *)webView willPerformDropWithSession:(id <UIDropSession>)session WK_API_AVAILABLE(ios(11.0));
 - (NSInteger)_webView:(WKWebView *)webView dataOwnerForDropSession:(id <UIDropSession>)session WK_API_AVAILABLE(ios(11.0));
 - (NSInteger)_webView:(WKWebView *)webView dataOwnerForDragSession:(id <UIDragSession>)session WK_API_AVAILABLE(ios(11.0));
+// FIXME (rdar://96239605): Expose these edit menu delegates as public API.
```
Comment (hortont424) [inline Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h]: Technically I think this should point at a non-opaque bugzilla bug (though I am guilty of sometimes pasting radars too)

**Review by hortont424 — APPROVED** [review]

File: Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h (hunk 1 above)
Comment (whsieh) [reply Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h]: Will do! Filed: https://bugs.webkit.org/show_bug.cgi?id=243102 (and I'll replace the rdar:// link with a bugzilla link).

File: Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h [hunk 2]
```diff
@@ -255,6 +256,9 @@ struct UIEdgeInsets;
...
+- (void)_webView:(WKWebView *)webView willPresentEditMenuWithAnimator:(id<UIEditMenuInteractionAnimating>)animator WK_API_AVAILABLE(ios(WK_IOS_TBA));
+- (void)_webView:(WKWebView *)webView willDismissEditMenuWithAnimator:(id<UIEditMenuInteractionAnimating>)animator WK_API_AVAILABLE(ios(WK_IOS_TBA));
```
Comment (megangardner) [inline Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:261]: #if HAVE(UI_EDIT_MENU_INTERACTION) around these?

**Review by megangardner — APPROVED** [review]

File: Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h (hunk 2 above)
Comment (whsieh) [reply Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:261]: No, since this is in a private header.

**Comment by whsieh** [comment]
Thanks for the reviews!

File: Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h (hunk 2 above)
Comment (hortont424) [reply Source/WebKit/UIProcess/API/Cocoa/WKUIDelegatePrivate.h:261]: No HAVE() in API/SPI headers, no.

**Pull request merged**